api/src/main/java/org/openmrs/api/handler/PersonSaveHandler.java

```diff
@@ -42,6 +42,9 @@ public void handle(Person person, User creator, Date dateCreated, String other)
 		// address collection
 		if (person.getAddresses() != null && person.getAddresses().size() > 0) {
 			for (PersonAddress pAddress : person.getAddresses()) {
+				if (pAddress.isBlank()){
```

Comment:
I would imagine that removeAddress and the call to PersonAddress.setPerson are technically mutually exclusive, so this should probably be an if-else clause or you should call continue after removing it. I know the ConceptSaveHandler code for descriptions has this same issue, i will fix it